formats.go

```diff
@@ -13,6 +13,8 @@ var (
 	nameRegex              = regexp.MustCompile(`^[a-zA-Z0-9][a-z0-9A-Z\. \-_]{2,128}$`)
 	codeRegex              = regexp.MustCompile("^([0-9abcdefghjkmnpqrtuvwxyz]{16}|[0-9]{6})$")
 	featureValueLabelRegex = regexp.MustCompile(`^[a-z0-9][a-z0-9-_\.]{1,128}$`)
+	annotationKey          = regexp.MustCompile(`^([^-\.\/][a-z0-9-\.\/]*[^-\.\/]|[a-z0-9-])$`)
```

Comment:
You can enforce the length limitations through your regex by using a length spec {2,80} will match {2,80} characters of th e previous pattern.
Also put your regex into debuggex ( https://www.debuggex.com/ ).
I think your OR | isn't working as intended?
![image](https://user-images.githubusercontent.com/1552848/48846127-a8df5d00-ed74-11e8-81e4-8aef05ae1170.png)
Also if you want to have groups, but don't want them to capture you can use (?:CONTENTS) , which is just like specifying a normal group, but leads with `?:` on the inside
They should perform a little faster, and make it easier to traverse the groups you do want.

Reply:
Also please add a suffix of Regex to these to follow the pattern, I got confused below when I read the code, because the regex name is so close to the type.